libafl_nyx/src/helper.rs

```diff
@@ -0,0 +1,129 @@
+/// [`NyxHelper`] is used to wrap `NyxProcess`
+use std::{
+    fmt::{self, Debug},
+    path::Path,
+    time::Duration,
+};
+
+use libafl::Error;
+use libnyx::{NyxProcess, NyxReturnValue};
+
+pub struct NyxHelper {
+    pub nyx_process: NyxProcess,
+    /// real size of trace_bits
+    pub real_map_size: usize,
+    // real size of the trace_bits
+    pub map_size: usize,
+    /// shared memory with instruction bitmaps
+    pub trace_bits: *mut u8,
+}
+
+const MAX_FILE: u32 = 1024 * 1024;
+#[derive(Clone, Copy, Debug)]
+pub enum NyxProcessType {
+    /// stand alone mode
+    ALONE,
+    /// parallel mode's parent, used to create snapshot
+    PARENT,
+    /// parallel mode's child, consume snapshot and execute
+    CHILD,
+}
+impl NyxHelper {
+    /// create `NyxProcess` and do basic settings
+    /// It will convert instance to parent or child using `parent_cpu_id` when set`parallel_mode`
+    pub fn new(
+        target_dir: &Path,
+        cpu_id: u32,
+        snap_mode: bool,
+        parallel_mode: bool,
+        parent_cpu_id: Option<u32>,
+    ) -> Result<Self, Error> {
+        let sharedir = match target_dir.to_str() {
+            Some(x) => x,
+            None => return Err(Error::illegal_argument("can't convert sharedir to str")),
+        };
+        let work_dir = target_dir.join("workdir");
+        let work_dir = work_dir.to_str().expect("unable to convert workdir to str");
+        let nyx_type = if parallel_mode {
+            let parent_cpu_id = match parent_cpu_id {
+                None => {
+                    return Err(Error::illegal_argument(
+                        "please set parent_cpu_id in nyx parallel mode",
```

Comment:
At some point it could still be nice to automate this, as discussed in https://github.com/nyx-fuzz/libnyx/issues/15